react-front-end/__tests__/tsrc/search/AdvancedSearch.test.tsx

```diff
@@ -285,11 +300,67 @@ describe("Rendering of wizard", () => {
     mockGetAdvancedSearchByUuid.mockResolvedValue(advancedSearchDefinition);
     const { container } = await renderAdvancedSearchPage();
 
-    // Collect all default values.
-    const defaultLabelsAndValues = mockedControls.map(([c, controlValue]) =>
-      getControlValue(container, Array.from(controlValue.keys()), c.controlType)
+    // Collect _all_ values.
+    const defaultLabelsAndValues = buildMockedControls().map(
+      ([c, controlValue]) =>
+        getControlValue(
+          container,
+          Array.from(controlValue.keys()),
+          c.controlType
+        )
     );
 
+    // Make sure the current values match the default
     expect(defaultLabelsAndValues).toEqual(mockedLabelsAndValues);
+
+    // Clear all input value
+    await clickClearButton(container);
+
+    const afterClearLabelsAndValues: WizardControlLabelValue[] = pipe(
+      buildMockedControls(),
+      A.map<MockedControlValue, WizardControlLabelValue>(([c, controlValue]) =>
+        pipe(
+          getControlValue(
+            container,
+            Array.from(controlValue.keys()),
+            c.controlType
+          ),
+          O.fromNullable,
+          O.getOrElse(() => new Map())
+        )
+      )
+    );
+
+    const isNotEmptyMap = (m: Map<unknown, unknown>): boolean => {
+      return !M.isEmpty(m);
+    };
```

Comment:
Not required (see following comment). But also, be aware this could have been done as:

```suggestion
    const isNotEmptyMap = (m: Map<unknown, unknown>): boolean => !M.isEmpty(m);
```